api/src/main/java/edu/wpi/first/shuffleboard/api/sources/AbstractDataSource.java

```diff
@@ -116,4 +117,13 @@ public boolean hasClients() {
     return !clients.isEmpty();
   }
 
+  @Override
+  public boolean equals(Object o) {
+    return o instanceof DataSource<?> && Objects.equals(((DataSource<?>) o).getId(), this.getId());
```

Comment:
Check that the classes are equal, not just instanceof the interface.  This might blow up on interactions with destroyed sources

Reply:
How do we want this to interact with destroyed sources?

I think they don't inherit from AbstractDataSource, so they wouldn't have this impl anyway. Should they?

Reply:
Ah, if they don't extend from this class then I'm fine with the PR as is

Reply:
I haven't looked too much at the DestroyedSource impl, but theoretically this comparison might return true if the rhs was a destroyed source with the same URI. If that's possible, is it what we want?

Reply:
Comparison with a destroyed source in rhs did indeed return true. Changed to check class, and added some tests.